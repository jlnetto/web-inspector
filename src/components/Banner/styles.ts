import styled from 'styled-components';

export const BannerContainer = styled.div<{ type: 'error' | 'success' }>`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 12px;
  background-color: ${({ type }) => (type === 'error' ? '#ffebee' : '#e8f5e9')};
  color: ${({ type }) => (type === 'error' ? '#b71c1c' : '#2e7d32')};
  gap: 5px;
`;

export const IconContainer = styled.div`
  display: flex;
  margin-right: 8px;
`;

export const Message = styled.p`
  margin: 0;
`;

export const CloseWrapper = styled.div`
  display: flex;
  align-self: baseline;
  cursor: pointer;
`;
