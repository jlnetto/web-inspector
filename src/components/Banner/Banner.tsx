import { ReactComponent as ErrorIcon } from '../../assets/error.svg';
import { ReactComponent as SuccessIcon } from '../../assets/success.svg';
import { ReactComponent as CloseIcon } from '../../assets/close.svg';
import { BannerContainer, IconContainer, Message, CloseWrapper } from './styles';

type BannerProps = {
  type: 'error' | 'success';
  message: string;
  onClose?: () => void;
};

const Banner: React.FC<BannerProps> = ({ type, message, onClose }) => {
  const IconComponent = type === 'error' ? ErrorIcon : SuccessIcon;

  return (
    <BannerContainer type={type}>
      <IconContainer>
        <IconComponent />
      </IconContainer>
      <Message>{message}</Message>
      <CloseWrapper onClick={onClose}>
        <CloseIcon data-testid="close" />
      </CloseWrapper>
    </BannerContainer>
  );
};

export default Banner;
