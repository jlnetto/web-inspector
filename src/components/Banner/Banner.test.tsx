import { render, fireEvent, screen } from '@testing-library/react';
import Banner from './Banner';

describe('Banner component', () => {
  test('renders with error type and message', () => {
    const { asFragment } = render(
      <Banner type="error" message="An error occurred" />
    );

    const errorMessage = screen.getByText('An error occurred');
    expect(errorMessage).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders with success type and message', () => {
    const { asFragment } = render(
      <Banner type="success" message="Success message" />
    );

    const successMessage = screen.getByText('Success message');
    expect(successMessage).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('calls onClose when close button is clicked', () => {
    const onCloseMock = jest.fn();
    render(
      <Banner type="error" message="An error occurred" onClose={onCloseMock} />
    );

    const closeButton = screen.getByTestId('close');
    fireEvent.click(closeButton);

    expect(onCloseMock).toHaveBeenCalled();
  });
});
