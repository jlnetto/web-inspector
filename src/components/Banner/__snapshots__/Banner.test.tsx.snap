// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders with error type and message 1`] = `
<DocumentFragment>
  <div
    class="sc-bgqQcB bVhXrd"
    type="error"
  >
    <div
      class="sc-gTRrQi jbNXEy"
    >
      <div />
    </div>
    <p
      class="sc-ewnqHT lmvpbT"
    >
      An error occurred
    </p>
    <div
      class="sc-fFGjHI emCJSz"
    >
      <div
        data-testid="close"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner component renders with success type and message 1`] = `
<DocumentFragment>
  <div
    class="sc-bgqQcB dAUjVg"
    type="success"
  >
    <div
      class="sc-gTRrQi jbNXEy"
    >
      <div />
    </div>
    <p
      class="sc-ewnqHT lmvpbT"
    >
      Success message
    </p>
    <div
      class="sc-fFGjHI emCJSz"
    >
      <div
        data-testid="close"
      />
    </div>
  </div>
</DocumentFragment>
`;
